### PR TITLE
test: add comprehensive unit tests for :core:datastore module

### PR DESCRIPTION
**Pull request by fpMiguel**
## Summary

Adds **92 JVM unit tests** to `:core:datastore` — the module previously had no direct
tests. Covers `PreferencesDataSource` (32), `PersistentDataDataSource` (48) and
`BackupMapper` (12) with edge-case coverage, and moves the existing datastore tests
from `:app` into the module where the code under test lives.

## Changes

| File | Change |
|------|--------|
| `core/datastore/build.gradle.kts` | Robolectric/JUnit test configuration + test dependencies |
| `gradle/libs.versions.toml` | Added `androidx.test:core` to the version catalog |
| `core/datastore/src/test/.../BackupMapperTest.kt` | 12 tests — `toSettingsBackup` mapping + `applyTo` integration |
| `core/datastore/src/test/.../PreferencesDataSourceTest.kt` | 32 tests — roundtrips, enum parsing, preference overrides, Flow emission |
| `core/datastore/src/test/.../PersistentDataDataSourceTest.kt` | 48 tests — setters, notification IDs, pending-snooze CRUD, data Flow |
| `app/src/test/.../location/` (deleted) | Removed — tests moved to `:core:datastore` where the code under test lives |

## Notes

- Test infrastructure: Robolectric + JUnit 4 + kotlinx-coroutines-test
- Per review feedback, default-value tests and tests that merely re-state Android
  `SharedPreferences` behavior were removed
- _Note: AI assistance was used in drafting this code and/or tests._

### CODE REVIEW

**Review by Futsch1 — APPROVED** [review]
@fpMiguel 
Thank you for your contribution! Your PR is still draft, so I cannot merge it yet. I will look into all of your PRs and review them. Once they are all approved and good to merge, I will do this all in one go to come to a working state of tests again.

Oh, and it would be great to disclose if you used AI to generate the PRs.

File: core/datastore/src/test/java/com/futsch1/medtimer/core/datastore/PersistentDataDataSourceTest.kt [hunk 1]
```diff
@@ -0,0 +1,493 @@
...
+        medTimerPrefs.edit().clear().commit()
+        dataSource = PersistentDataDataSource(
+            defaultPrefs,
+            medTimerPrefs,
+            CoroutineScope(Dispatchers.Unconfined),
+            GsonBuilder().create()
+        )
+    }
+
+    // ── A: Default values ──────────────────────────────────────────────
```
Comment (Futsch1) [inline core/datastore/src/test/java/com/futsch1/medtimer/core/datastore/PersistentDataDataSourceTest.kt]: Do not test default values. This is trivial and creates unnecessary work when changing these values.

File: core/datastore/src/test/java/com/futsch1/medtimer/core/datastore/PreferencesDataSourceTest.kt [hunk 2]
```diff
@@ -0,0 +1,498 @@
...
+            .getSharedPreferences("test_prefs_ds", Context.MODE_PRIVATE)
+        sharedPreferences.edit().clear().commit()
+        dataSource = PreferencesDataSource(
+            sharedPreferences,
+            CoroutineScope(Dispatchers.Unconfined),
+            GsonBuilder().create()
+        )
+    }
+
+    // ── A: Default values ──────────────────────────────────────────────
```
Comment (Futsch1) [inline core/datastore/src/test/java/com/futsch1/medtimer/core/datastore/PreferencesDataSourceTest.kt]: Same here - no default values

**Review by Futsch1 — CHANGES_REQUESTED** [review]

File: core/datastore/build.gradle.kts [hunk 3]
```diff
@@ -32,4 +43,10 @@ dependencies {
     implementation(libs.androidx.core.ktx)
     implementation(libs.androidx.preference.ktx)
     coreLibraryDesugaring(libs.desugar.jdk.libs)
+
+    testImplementation(libs.junit4)
+    testImplementation(kotlin("test-junit"))
+    testImplementation(libs.robolectric)
+    testImplementation(libs.kotlinx.coroutines.test)
+    testImplementation(libs.androidx.test.core)
```
Comment (Futsch1) [inline core/datastore/build.gradle.kts:51]: Are you sure you need to add libs.androidx.test.core explicitly? Other modules did not need it for running unit tests.

Comment (fpMiguel) [reply core/datastore/build.gradle.kts:51]: Other modules don't need it because their test classpaths include androidx.compose.ui.test.junit4 or hilt-android-testing, which bring androidx.test:core along transitively; :core:datastore has neither, so an explicit declaration is required here for:

- BackupMapperTest.kt — uses ApplicationProvider (and SharedPreferences.edit().commit())
- PersistentDataDataSourceTest.kt
- PreferencesDataSourceTest.kt

**Review by Futsch1 — APPROVED** [review]
